### PR TITLE
feat:[SSCA-1225]: Added implementation of new artifact listing api

### PR DESCRIPTION
**Pull request by arpit-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/57389)

### CODE REVIEW

**Comment by arpit-harness** [comment]
trigger uts

**Comment by arpit-harness** [comment]
trigger messagecheck

**Comment by arpit-harness** [comment]
trigger uts

**Comment by arpit-harness** [comment]
trigger uts

**Review by inderschera — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57389)* status: 11 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`ssca-manager/service/src/main/java/io/harness/ssca/api/ArtifactApiImpl.java` line 67 at r3](https://reviewable.io/reviews/harness/harness-core/57389#-NnmaV51DjxlMxT5iR82:-NnmaV51DjxlMxT5iR83:b-o0mwd4) ([raw file](https://github.com/harness/harness-core/blob/b1ed134c95e9c666133292341eeda2260e3ce574/ssca-manager/service/src/main/java/io/harness/ssca/api/ArtifactApiImpl.java#L67)):*
> ```Java
>     Pageable pageable = PageResponseUtils.getPageable(page, limit, sort, order);
>     Page<ArtifactListingResponseV2> artifactEntities =
>         artifactService.listArtifacts(harnessAccount, org, project, body, pageable, "image");
> ```

Has migration already gone in production for adding type? If it is going in this release, and say if background migration failed for some reason, then won't this cause issues on artifact listing page?
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/api/ArtifactApiUtils.java` line 26 at r3](https://reviewable.io/reviews/harness/harness-core/57389#-NnmZuXABNyaCbENwwGH:-NnmZuXABNyaCbENwwGI:blqtx7g) ([raw file](https://github.com/harness/harness-core/blob/b1ed134c95e9c666133292341eeda2260e3ce574/ssca-manager/service/src/main/java/io/harness/ssca/api/ArtifactApiUtils.java#L26)):*
> ```Java
> 
> @Slf4j
> public class ArtifactApiUtils {
> ```

nit: should we name it to ArtifactApiMapper?
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/api/ArtifactApiUtils.java` line 47 at r3](https://reviewable.io/reviews/harness/harness-core/57389#-Nnm_BhBAjgTCQp4k-km:-Nnm_BhBAjgTCQp4k-kn:bpy1mze) ([raw file](https://github.com/harness/harness-core/blob/b1ed134c95e9c666133292341eeda2260e3ce574/ssca-manager/service/src/main/java/io/harness/ssca/api/ArtifactApiUtils.java#L47)):*
> ```Java
>   }
> 
>   public static Page<ArtifactListingResponse> artifactListingResponseMapper(
> ```

nit: rename to toArtifactListingResponseList()
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/api/ArtifactApiUtils.java` line 56 at r3](https://reviewable.io/reviews/harness/harness-core/57389#-Nnm_Q042vVdAuMQzixb:-Nnm_Q042vVdAuMQzixc:b-1o21x) ([raw file](https://github.com/harness/harness-core/blob/b1ed134c95e9c666133292341eeda2260e3ce574/ssca-manager/service/src/main/java/io/harness/ssca/api/ArtifactApiUtils.java#L56)):*
> ```Java
>   }
> 
>   private static ArtifactListingResponse covertToV1Response(ArtifactListingResponseV2 responseV2) {
> ```

nit: rename to toArtifactListingResponse
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/search/SearchServiceImpl.java` line 235 at r3](https://reviewable.io/reviews/harness/harness-core/57389#-Nnm_kOl4I1uCCaJwBcq:-Nnm_kOl4I1uCCaJwBcr:b-29odu9) ([raw file](https://github.com/harness/harness-core/blob/b1ed134c95e9c666133292341eeda2260e3ce574/ssca-manager/service/src/main/java/io/harness/ssca/search/SearchServiceImpl.java#L235)):*
> ```Java
>     SearchRequest searchRequest = SearchRequest.of(s
>         -> s.index(index)
>                .query(ArtifactQueryBuilder.getQuery(accountId, orgIdentifier, projectIdentifier, type, filter))
> ```

did we do migration for elk as well? If not, won't this cause issue?
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/services/ArtifactServiceImpl.java` line 805 at r3](https://reviewable.io/reviews/harness/harness-core/57389#-Nnma32hEXlLm_EvpkJ0:-Nnma32i4KmtlQVRZqwN:b-ik5s94) ([raw file](https://github.com/harness/harness-core/blob/b1ed134c95e9c666133292341eeda2260e3ce574/ssca-manager/service/src/main/java/io/harness/ssca/services/ArtifactServiceImpl.java#L805)):*
> ```Java
> 
>   private String getVariantType(ArtifactEntity artifact) {
>     if (artifact.getType().equals("image")) {
> ```

is getType() not an enum?
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/services/ArtifactServiceImpl.java` line 810 at r3](https://reviewable.io/reviews/harness/harness-core/57389#-NnmaFVM4Uf12AWtgbFe:-NnmaFVM4Uf12AWtgbFf:b-l61vr2) ([raw file](https://github.com/harness/harness-core/blob/b1ed134c95e9c666133292341eeda2260e3ce574/ssca-manager/service/src/main/java/io/harness/ssca/services/ArtifactServiceImpl.java#L810)):*
> ```Java
>       RepositoryArtifactSpec spec = (RepositoryArtifactSpec) artifact.getSpec();
>       if (spec.getBranch() != null) {
>         return "branch";
> ```

Why are we returning here? Can't we have multiple of these populated?
For example, say we want to show branch and commitsha on UI?

File: ssca-manager/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -345,23 +345,26 @@ paths:
...
-            default: 'false'
+            type: string
+            enum:
+              - all
+              - latest
+            default: all
           in: query
-          name: latest_only
-          description: Flag used to for toggling to latest artifacts view
+          name: mode
```
Comment (inderschera) [inline ssca-manager/contracts/openapi/v1/openapi.yaml:360]: Does view/list_view sound better here than mode?

**Review by inderschera — COMMENTED** [review]
> Dismissed @inderschera from a discussion.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57389)* status: 11 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`ssca-manager/service/src/main/java/io/harness/ssca/services/ArtifactServiceImpl.java` line 810 at r3](https://reviewable.io/reviews/harness/harness-core/57389#-NnmaFVM4Uf12AWtgbFe:-NnmcXq0-_0Phzw9Eze3:bte65fw) ([raw file](https://github.com/harness/harness-core/blob/b1ed134c95e9c666133292341eeda2260e3ce574/ssca-manager/service/src/main/java/io/harness/ssca/services/ArtifactServiceImpl.java#L810)):*
<details><summary><i>Previously, inderschera (Inderpreet Chera) wrote…</i></summary><blockquote>

Why are we returning here? Can't we have multiple of these populated?
For example, say we want to show branch and commitsha on UI?
</blockquote></details>

Saw the YAML proposed. We could only have either of branch, commitsha or gitTag. Combination is not allowed. So, we should be good here.

File: ssca-manager/contracts/openapi/v1/openapi.yaml (hunk 1 above)
Comment (arpit-harness) [reply ssca-manager/contracts/openapi/v1/openapi.yaml:360]: i can update it to view_mode.

**Review by arpit-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57389)* status: 11 file(s) to review. 1 change requests. 4. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`ssca-manager/service/src/main/java/io/harness/ssca/api/ArtifactApiImpl.java` line 67 at r3](https://reviewable.io/reviews/harness/harness-core/57389#-NnmaV51DjxlMxT5iR82:-Nnn1lCk7f1h-PS4WNnd:b-8q8w6r) ([raw file](https://github.com/harness/harness-core/blob/b1ed134c95e9c666133292341eeda2260e3ce574/ssca-manager/service/src/main/java/io/harness/ssca/api/ArtifactApiImpl.java#L67)):*
<details><summary><i>Previously, inderschera (Inderpreet Chera) wrote…</i></summary><blockquote>

Has migration already gone in production for adding type? If it is going in this release, and say if background migration failed for some reason, then won't this cause issues on artifact listing page?
</blockquote></details>

existing artifacts already have "image" only few faulty records had "repo/image". If migration fails then those records won't show up.
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/api/ArtifactApiUtils.java` line 26 at r3](https://reviewable.io/reviews/harness/harness-core/57389#-NnmZuXABNyaCbENwwGH:-Nnn23W-781dsfXTnXJe:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/b1ed134c95e9c666133292341eeda2260e3ce574/ssca-manager/service/src/main/java/io/harness/ssca/api/ArtifactApiUtils.java#L26)):*
<details><summary><i>Previously, inderschera (Inderpreet Chera) wrote…</i></summary><blockquote>

nit: should we name it to ArtifactApiMapper?
</blockquote></details>

Done
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/api/ArtifactApiUtils.java` line 47 at r3](https://reviewable.io/reviews/harness/harness-core/57389#-Nnm_BhBAjgTCQp4k-km:-Nnn26jK3Zu9d_N3O--s:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/b1ed134c95e9c666133292341eeda2260e3ce574/ssca-manager/service/src/main/java/io/harness/ssca/api/ArtifactApiUtils.java#L47)):*
<details><summary><i>Previously, inderschera (Inderpreet Chera) wrote…</i></summary><blockquote>

nit: rename to toArtifactListingResponseList()
</blockquote></details>

Done
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/api/ArtifactApiUtils.java` line 56 at r3](https://reviewable.io/reviews/harness/harness-core/57389#-Nnm_Q042vVdAuMQzixb:-Nnn27nrCBWuW16wD9Au:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/b1ed134c95e9c666133292341eeda2260e3ce574/ssca-manager/service/src/main/java/io/harness/ssca/api/ArtifactApiUtils.java#L56)):*
<details><summary><i>Previously, inderschera (Inderpreet Chera) wrote…</i></summary><blockquote>

nit: rename to toArtifactListingResponse
</blockquote></details>

Done
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/search/SearchServiceImpl.java` line 235 at r3](https://reviewable.io/reviews/harness/harness-core/57389#-Nnm_kOl4I1uCCaJwBcq:-Nnn2A5J0yVHcp0MDUeb:bny4wg3) ([raw file](https://github.com/harness/harness-core/blob/b1ed134c95e9c666133292341eeda2260e3ce574/ssca-manager/service/src/main/java/io/harness/ssca/search/SearchServiceImpl.java#L235)):*
<details><summary><i>Previously, inderschera (Inderpreet Chera) wrote…</i></summary><blockquote>

did we do migration for elk as well? If not, won't this cause issue?
</blockquote></details>

yes we did.
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/services/ArtifactServiceImpl.java` line 805 at r3](https://reviewable.io/reviews/harness/harness-core/57389#-Nnma32hEXlLm_EvpkJ0:-Nnn2DZ8B13zn6cOWT0f:bt6kxv3) ([raw file](https://github.com/harness/harness-core/blob/b1ed134c95e9c666133292341eeda2260e3ce574/ssca-manager/service/src/main/java/io/harness/ssca/services/ArtifactServiceImpl.java#L805)):*
<details><summary><i>Previously, inderschera (Inderpreet Chera) wrote…</i></summary><blockquote>

is getType() not an enum?
</blockquote></details>

No, because of the faulty records. If migration failed then it could break artifact listing. We can make in enum in next stable release.
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/services/ArtifactServiceImpl.java` line 810 at r3](https://reviewable.io/reviews/harness/harness-core/57389#-NnmaFVM4Uf12AWtgbFe:-Nnn2T7-1ZuVWsxJ_8F7:b2nhb2c) ([raw file](https://github.com/harness/harness-core/blob/b1ed134c95e9c666133292341eeda2260e3ce574/ssca-manager/service/src/main/java/io/harness/ssca/services/ArtifactServiceImpl.java#L810)):*
<details><summary><i>Previously, inderschera (Inderpreet Chera) wrote…</i></summary><blockquote>

Saw the YAML proposed. We could only have either of branch, commitsha or gitTag. Combination is not allowed. So, we should be good here.
</blockquote></details>

Ack

**Review by arpit-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57389)* status: 11 file(s) to review. 1 change requests. 4. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`ssca-manager/service/src/main/java/io/harness/ssca/api/ArtifactApiUtils.java` line 26 at r3](https://reviewable.io/reviews/harness/harness-core/57389#-NnmZuXABNyaCbENwwGH:-Nnn3w7HEqOYNqL3jtL7:b-ghlg4b) ([raw file](https://github.com/harness/harness-core/blob/b1ed134c95e9c666133292341eeda2260e3ce574/ssca-manager/service/src/main/java/io/harness/ssca/api/ArtifactApiUtils.java#L26)):*
<details><summary><i>Previously, arpit-harness wrote…</i></summary><blockquote>

Done
</blockquote></details>

AritfactApiUtils is fine for now.

**Review by mankrit-singh — APPROVED** [review]

**Comment by arpit-harness** [comment]
trigger uts

**Review by inderschera — CHANGES_REQUESTED** [review]
> Reviewed 5 of 8 files at r1, 2 of 2 files at r3, 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57389)* status: 1 file(s) to review. 1 change requests. 

___
*[`ssca-manager/contracts/openapi/v1/openapi.yaml` line 360 at r3](https://reviewable.io/reviews/harness/harness-core/57389#gh-1447074043:-NnnFuWf3zJgJAyXRA0X:b-alxkn) ([raw file](https://github.com/harness/harness-core/blob/b1ed134c95e9c666133292341eeda2260e3ce574/ssca-manager/contracts/openapi/v1/openapi.yaml#L360)):*
<details><summary><i>Previously, arpit-harness wrote…</i></summary><blockquote>

i can update it to view_mode.
</blockquote></details>

yes, that sounds better
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/services/ArtifactServiceImpl.java` line 805 at r4](https://reviewable.io/reviews/harness/harness-core/57389#-NnnGHPL5S95kWfkFdKw:-NnnGHPL5S95kWfkFdKx:bpcgnk9) ([raw file](https://github.com/harness/harness-core/blob/6e47e1340e2afd4c1bdaba31df52b874a7acd0c8/ssca-manager/service/src/main/java/io/harness/ssca/services/ArtifactServiceImpl.java#L805)):*
> ```Java
> 
>   private String getVariantType(ArtifactEntity artifact) {
>     if (artifact.getType().equals("image")) {
> ```

Can artifact.getType() be null? To avoid NPE probably have "image".equals(artifact.getType())

**Review by inderschera — APPROVED** [review]
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57389)* status: <span>:shipit:</span> complete! ✓

**Review by mankrit-singh — APPROVED** [review]

**Comment by arpit-harness** [comment]
trigger ut4 ut6 ut9

**Comment by arpit-harness** [comment]
trigger ut4

**Pull request merged**